src/js/files/components/Toolbar.tsx

```diff
@@ -0,0 +1,65 @@
+import { forEach } from "lodash-es";
+import React from "react";
+import { connect } from "react-redux";
+import { useCheckAdminRoleOrPermission } from "../../administration/hooks";
+import { Alert, Icon, UploadBar } from "../../base";
+import { Permission } from "../../groups/types";
+import { createRandomString } from "../../utils/utils";
+import { upload } from "../actions";
+
+type UploadToolbarProps = {
+    onDrop: (fileType: string, acceptedFiles: File[]) => void;
+    fileType: string;
+    tip?: string;
+    message?: React.ReactNode;
+    /* For validating file type */
+    validationRegex?: RegExp;
+};
+
+/*
+ * Renders an UploadBar if the user has permission to upload files.
+ */
+export function UploadToolbar({ onDrop, fileType, message, validationRegex }: UploadToolbarProps) {
+    const { hasPermission: canCreate } = useCheckAdminRoleOrPermission(Permission.create_ref);
```

Comment:
You sure it makes sense to check the ability to create a reference? I would assume the checking for upload permissions makes much more sense here

Reply:
Yes, good catch - I was basing this off of code from other components